### DISCUSSION TITLE
Filter Results Based on Originating Site

### DISCUSSION THREAD

**Discussion by cleve-virtuas** [microsoft-search/pnp-modern-search#2698] [Q&A]
A large number of sites feed into a single custom search page and we want the results to be scoped to the sites where the searches originated. The issue we are hitting is with referencing the tokens passed via URL and maintaining that filter or scope during use.
I have the GUID and the sitename added to the URL like so:
sharepoint.com/sites/sitename/SitePages/search-page.aspx?guid=XXXXXXX?site=sitename

I added references to the tokens to change the scope, but that did not work. I tried instead to filter the results and have had no luck, Any help or direction here would be greatly appreciated.

**Comment by kasperbolarsen** [comment]
I guess there is a typo in the URL, sharepoint.com/sites/sitename/SitePages/search-page.aspx?guid=XXXXXXX?site=sitename should be sharepoint.com/sites/sitename/SitePages/search-page.aspx?guid=XXXXXXX**&**site=sitename right ?

**Comment by cleve-virtuas** [comment]
I used only one query string and it did the same thing. I also altered the URL to use & and it had no effect.
The other part of the problem is that once they search on the page, this query string is lost.

**Reply by wobba** [reply]
Can you share how you are picking up the URL parameter in the query template section? And maybe a trace of the query being made, which should have the token replaced when looking at it.

**Comment by kasperbolarsen** [comment]
Hi @cleve-virtuas 
Perhaps starting with a very simple scenario and extend from there will work? I got it working doing this: ( version 4.8.0)
1)pick SharePoint as Data source
2) go to page 3 
![image](https://user-images.githubusercontent.com/20593570/213023255-4b419aaf-4664-4959-8c0f-e0db91f6f43c.png)

I have to update the URL of the page with ?ContentTypeId=whatever  in order to select that property in "queryParameter's properties

Now you should be able to hit the page with a URL like [PageURL]?ContentTypeId=0x0101009D1CB255DA76424F860D91F20E6C4118001A2BCBDB1A255E4CA428D5D05700C779  and see all the Site Pages that the current user has access to :-)

Does this work for you ?

**Comment by kasperbolarsen** [comment]
But perhaps this will fit your request ( two or more query parameters in the URL ) even better ?
No updates on page 3, only in "Query template" on page 1
![image](https://user-images.githubusercontent.com/20593570/213027305-834ef2f7-9a00-4541-93e7-e1dd9b46157f.png)
